src/coverlet.core/Coverage.cs

```diff
@@ -197,6 +196,31 @@ private void CalculateCoverage()
                     }
                 }
 
+                // for MoveNext() compiler autogenerated method we need to patch false positive (IAsyncStateMachine for instance) 
+                // we'll remove all MoveNext() not covered branch              
+                foreach (var document in result.Documents)
+                {
+                    List<KeyValuePair<(int, int), Branch>> branchesToRemove = new List<KeyValuePair<(int, int), Branch>>();
+                    foreach (var branch in document.Value.Branches)
+                    {
+                        //if one branch is covered we search the other one only if it's not covered
+                        if (CecilSymbolHelper.IsMoveNext(branch.Value.Method) && branch.Value.Hits > 0)
+                        {
+                            foreach (var moveNextBranch in document.Value.Branches)
+                            {
+                                if (moveNextBranch.Value.Method == branch.Value.Method && moveNextBranch.Value != branch.Value && moveNextBranch.Value.Hits == 0)
+                                {
+                                    branchesToRemove.Add(moveNextBranch);
+                                }
+                            }
+                        }
+                    }
+                    foreach (var branchToRemove in branchesToRemove)
+                    {
+                        document.Value.Branches.Remove(branchToRemove.Key);

```

Comment:
on next runtime version we can remove directly during enumeration(`Dictionary<(int Line, int Ordinal), Branch>`) https://github.com/dotnet/coreclr/pull/18854 https://github.com/dotnet/corefx/issues/31459

Reply:
I'm very interested in helping to get this pr merged - @MarcoRossignoli do you need to update this pr now that they released a new version of the clr last week?

Reply:
@abe545 thank's for the interest. I'll verify if above amend has been released, however it's not essential to this PR. When @tonerdo will be able to review we'll know if it's good to merge or not!

Reply:
NB. It's not my intention put pressure on maintainers!@tonerdo take all time you need, and thank's a lot for your work!

Reply:
Would love to see this merged. Anything holding this back, @tonerdo ?